src/common/hostname.cc

```diff
@@ -14,12 +14,20 @@
 
 #include "common/hostname.h"
 
+#include <iostream>
 #include <unistd.h>
 
 std::string ceph_get_hostname()
 {
-  char buf[1024];
-  gethostname(buf, 1024);
+  std::size_t max_char_length = 1024;
+  char buf[max_char_length];
+  int r = gethostname(buf, sizeof(buf));
+
+  if (r < 0) {
+    std::cerr << "invalid hostname or hostname is too long" << std::endl;
```

Comment:
i don't think the stderr is always available when this function gets called. for example, `md_config_t::expand_meta()` can be called by a daemonized daemon. `md_config_t::expand_meta()` calls `ceph_get_hostname()` indirectly.